src/Symfony/Component/HttpKernel/HttpCache/HttpCache.php

```
@@ -219,7 +219,11 @@ public function handle(Request $request, int $type = HttpKernelInterface::MAIN_R
             $this->record($request, 'reload');
             $response = $this->fetch($request, $catch);
         } else {
-            $response = $this->lookup($request, $catch);
+            try {
+                $response = $this->lookup($request, $catch);
+            } catch (CacheWasLockedException) {
+                $response = $this->lookup($request, $catch);
+            }
```

This allows to restart the process only once. 

In fact, the old code would wait for the lock to be released, but only try to acquire it later on. That could, in theory, allow for repeated cycles without actually reaching the 503 condition in `lock()`. 

Maybe it is even better this way to fail with the exception instead of silenty retrying over and over again?